subprojects/core/src/integTest/groovy/org/gradle/api/tasks/CacheTaskArchiveErrorIntegrationTest.groovy

```diff
@@ -218,7 +220,7 @@ class CacheTaskArchiveErrorIntegrationTest extends AbstractIntegrationSpec {
         when:
         fails "producesLink"
         then:
-        failureHasCause("Failed to store cache entry for task ':producesLink'")
+        failureHasCause(matchesRegexp("Failed to store cache entry [0-9a-z]+ for task ':producesLink': Could not pack tree 'outputDirectory'.*"))
```

Comment:
I don't see a reason why we shouldn't test the whole message here or with the other three problems where we only check the prefix.

Reply:
done